### PR TITLE
feature/1107_general_encoding

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #1107 
* 100% tests passed:
```
Tests run: 194, Failures: 0, Errors: 0, Skipped: 0
```

### CODE REVIEW

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSICharsets.java [hunk 1]
```diff
@@ -30,19 +30,153 @@ private NGSICharsets() {
     } // NGSICharsets
     
     /**
-     * Encodes a string for CartoDB.
+     * Encodes a string for PostgreSQL. This includes CartoDB. Only lowercase alphanumercics and _ are allowed.
```
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSICharsets.java]: alphanumercics -> alphanumerics

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIKafkaSinkTest.java [hunk 2]
```diff
@@ -130,13 +130,13 @@ public void testBuildTopicNameDmByServicePath() throws Exception {
     public void testBuildTopicNameDmByServicePathWithSlashServicePath() throws Exception {
         System.out.println(getTestTraceHead("[OrionKafkaSink.buildTopicName]") + "-------- When the "
                 + "root service-path is notified/defaulted and data_model=dm-by-service-path, "
-                + "the Kafka topic name is <service>");
+                + "the Kafka topic name is the concatenatin of <service> and <service-path>");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIKafkaSinkTest.java]: concatenatin -> concatenation

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 3]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+        } // try catch // try catch
+    } // testConfigureCollectionPrefixIsEncodedOldEncoding
+    
+    /**
+     * [NGSIMongoSink.configure] -------- Configured 'collection_prefix' is encoded when having forbiden characters.
+     */
+    @Test
+    public void testConfigureCollectionPrefixIsEncodedNewEncoding() {
+        System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                + "-------- Configured 'collection_prefix' is encoded when having forbiden characters");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbiden -> forbidden

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 4]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+        String collectionPrefix = "this\\is/a$prefix.with-forbiden,chars:-.";
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbidden too :no_entry_sign: :smile:

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 5]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+    } // testConfigureCollectionPrefixIsEncodedNewEncoding
     
     /**
      * [NGSIMongoSink.configure] -------- Configured 'db_prefix' is encoded when having forbiden characters.
      */
     @Test
-    public void testConfigureDBPrefixIsEncoded() {
+    public void testConfigureDBPrefixIsEncodedOldEncoding() {
         System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                 + "-------- Configured 'db_prefix' is encoded when having forbiden characters");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbidden too :no_entry_sign: :smile:

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 6]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
         String collectionPrefix = "sth_";
         String dbPrefix = "this\\is/a$prefix.with forbiden\"chars:-,";
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbidden too :no_entry_sign: :smile:

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 7]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+        } // try catch // try catch
+    } // testConfigureDBPrefixIsEncodedOldEncoding
+    
+    /**
+     * [NGSIMongoSink.configure] -------- Configured 'db_prefix' is encoded when having forbiden characters.
+     */
+    @Test
+    public void testConfigureDBPrefixIsEncodedNewEncoding() {
+        System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                + "-------- Configured 'db_prefix' is encoded when having forbiden characters");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbidden too :no_entry_sign: :smile:

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 8]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+        String collectionPrefix = "sth_";
+        String dbPrefix = "this\\is/a$prefix.with forbiden\"chars:-,";
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: forbidden too :no_entry_sign: :smile:

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 9]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+        
+        try {
+            String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
+                    isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_/";
+            
+            try {
+                assertEquals(expectedCollectionName, collectionName);
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 10]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
+            } catch (AssertionError e) {
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of 'sth_/'");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 11]
```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
...
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002f";
             
             try {
-                assertEquals("sth_/", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 12]
```diff
@@ -168,37 +282,89 @@ public void testBuildCollectionNameDMByServicePathRootServicePath() {
...
+        
+        try {
+            String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
+                    isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_/_someId_someType";
+            
+            try {
+                assertEquals(expectedCollectionName, collectionName);
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 13]
```diff
@@ -168,37 +282,89 @@ public void testBuildCollectionNameDMByServicePathRootServicePath() {
...
+            } catch (AssertionError e) {
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of "
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 14]
```diff
@@ -168,37 +282,89 @@ public void testBuildCollectionNameDMByServicePathRootServicePath() {
...
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002fxffffsomeIdxffffsomeType";
             
             try {
-                assertEquals("sth_/_someId_someType", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 15]
```diff
@@ -212,38 +378,91 @@ public void testBuildCollectionNameDMByEntityRootServicePath() {
...
+        
+        try {
+            String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
+                    isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_/_someId_someType_someName";
+            
+            try {
+                assertEquals(expectedCollectionName, collectionName);
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 16]
```diff
@@ -212,38 +378,91 @@ public void testBuildCollectionNameDMByEntityRootServicePath() {
...
+            } catch (AssertionError e) {
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of "
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java [hunk 17]
```diff
@@ -212,38 +378,91 @@ public void testBuildCollectionNameDMByEntityRootServicePath() {
...
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002fxffffsomeIdxffffsomeTypexffffsomeName";
             
             try {
-                assertEquals("sth_/_someId_someType_someName_someType", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: crated -> created

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java [hunk 18]
```diff
@@ -36,237 +36,1616 @@ public NGSICharsetsTest() {
...
+        } // try catch
+    } // testEncodeMySQLxffff
+    
+    /**
+     * [NGSICharsets.encodeMySQL] -------- Non lower case alphanumerics nor undersocre are encoded.
+     */
+    @Test
+    public void testEncodeMySQLRare() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                + "-------- Non lower case alphanumerics and undersocre are encoded");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: undersocre -> underscore

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java [hunk 19]
```diff
@@ -36,237 +36,1616 @@ public NGSICharsetsTest() {
...
+        } // try catch
+    } // testEncodeMongoDBDatabasexffff
+    
+    /**
+     * [NGSICharsets.encodeMongoDBDatabase] -------- Forbidden characters are encoded.
+     */
+    @Test
+    public void testEncodeMongoDBDatabaseForbiddenCharacters() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBDatabase]")
+                + "-------- Forbidde characters are encoded");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: Forbidde -> Forbidden

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java [hunk 20]
```diff
@@ -36,237 +36,1616 @@ public NGSICharsetsTest() {
...
+        } // try catch
+    } // testEncodeMongoDBCollectionxffff
+    
+    /**
+     * [NGSICharsets.encodeMongoDBCollection] -------- Forbidden characters are encoded.
+     */
+    @Test
+    public void testEncodeMongoDBCollectionForbiddenCharacters() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                + "-------- Forbidde characters are encoded");
```
Comment (pcoello25) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: Forbidde -> Forbidden

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_dynamodb_sink.md [hunk 21]
```diff
@@ -237,6 +238,21 @@ Regarding the amount of information per write, please read carefully [this](http
...
+####<a name="section2.3.5"></a>About the encoding
+Cygnus applies this specific encoding tailored to DynamoDB data structures:
+
+* Alphanumeric characters are not encoded.
+* Numeric characters are not encoded.
+* Underscore character, `_`, is not encoded.
+* Hyphen character, `-`, is not encoded.
+* Dot character, `.`, is not encoded.
+* Equals character, `=`, is encoded as `xffff`.
+* All other characters, including the slash in the FIWARE service paths, are encoded as a `x` character followed by the [Unicode](http://unicode-table.com) of the character.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_dynamodb_sink.md]: paths -> path¿?

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_kafka_sink.md [hunk 22]
```diff
@@ -42,19 +43,19 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
...
+* Data model by entity (`data_model=dm-by-entity`). For each entity, the notified/default FIWARE service path is concatenated to the notified entity ID and type in order to compose the topic name. If the FIWARE service path is the root one (`/`) then only the entity ID and type are concatenated.
+* Data model by attribute (`data_model=dm-by-attribute`). For each entity's attribute, the notified/default FIWARE service path is concatenated to the notified entity ID and type and to the notified attribute name in order to compose the topic name. If the FIWARE service path is the root one (`/`) then only the entity ID and type and the attribute name and type are concatenated.
 
-It must be said there is no known character set accepted and/or forbidden for Kafka.
+It must be said there is no known character set accepted and/or forbiden for Kafka. Anyway, certaing [encoding](#section2.3.2) is applied.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_kafka_sink.md]: forbiden -> forbidden

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_dynamodb_sink.md (hunk 21 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_dynamodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_kafka_sink.md (hunk 22 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_kafka_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java (hunk 20 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java (hunk 19 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java (hunk 18 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 17 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 16 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 15 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 14 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 13 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 12 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 11 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 10 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 9 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 8 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 7 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 6 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 5 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 4 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java (hunk 3 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIKafkaSinkTest.java (hunk 2 above)
Comment (frbattid) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIKafkaSinkTest.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSICharsets.java (hunk 1 above)
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSICharsets.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4

**Comment by pcoello25** [comment]
LGTM

**Pull request merged**